snapshot/bazel/fs_util_test.go

```diff
@@ -0,0 +1,178 @@
+package bazel
+
+import (
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/twitter/scoot/common/log/hooks"
+	"github.com/twitter/scoot/os/temp"
+)
+
+func init() {
+	log.AddHook(hooks.NewContextHook())
+}
+
+func makeTestingFiler() *bzFiler {
+	localStore, err := temp.TempDirDefault()
+	if err != nil {
+		log.Fatal(err)
+	}
+	localStorePathFn := func(bc *bzCommand) {
+		bc.localStorePath = localStore.Dir
+	}
+	bf := MakeBzFilerWithOptions(localStorePathFn)
+	return bf
+}
+
+// directory save tests
+
+func TestSaveEmptyDir(t *testing.T) {
+	root, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatal(err)
+	}
+	bf := makeTestingFiler()
+	id, err := bf.Ingest(root.Dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if id != emptyID {
+		t.Fatalf("Expected id to be %s, was %s", id, emptyID)
+	}
+}
+
+func TestSaveDir(t *testing.T) {
+	root, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmpDirName := "tmp"
+	tmpDir, err := root.FixedDir(tmpDirName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f, err := tmpDir.TempFile("nonempty")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = f.WriteString("some words\n")
+	if err != nil {
+		t.Fatal(err)
+	}
+	bf := makeTestingFiler()
+	id, err := bf.Ingest(root.Dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if id == emptyID {
+		t.Fatalf("Expected id to not be %s, was %s", emptyID, id)
+	}
+	size, err := getSize(id)
```

Comment:
You should be able to assert on the actual expected size and sha here, rather than just that they're non-empty?

Handy hint for manually doing this kind of thing, to generate a binary serialized proto you can run something like:
```
echo "files: {
  name: 'cheese'
  digest: {
    hash: '727d05cc3cb7b48eadc75bbbb8d740a9321691745f136d58270b93ba4f88c218'
    size_bytes: 19
  }
}
" | protoc --encode=google.devtools.remoteexecution.v1test.Directory google/devtools/remoteexecution/v1test/remote_execution.proto --protopath=/path/to/googleapis
```

which you can then pipe to shasum or wc to get shas and sizes.

Reply:
cool, thanks for the tip